salt/modules/sysmod.py

```diff
@@ -65,9 +65,10 @@ def doc(*args):
     '''
     docs = {}
     if not args:
-        for fun in __salt__:
-            docs[fun] = __salt__[fun].__doc__
-        return _strip_rst(docs)
+        funcs = __salt__.keys()
```

Comment:
Is there a specific reason for the switch to ``.keys()`` here? On Python 3 it is a dict_keys view, so no issue there, but on Python 2 it allocates a new list in memory. IIRC, simply iterating over a dict (like it was doing before) doesn't allocate a new list.